src/mpcCoreKit.ts

```diff
@@ -1044,6 +925,12 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
       } catch (err) {
         log.error("error initializing tkey with hashed share", err);
       }
+    } else {
```

Comment:
If the factorkey is invalid, it is already updated the metadata with "SHARE_DELETED"

Reply:
not necessarily.
if `checkIfFactorKeyValid == false`, then we have any of the following:
- `!factorKeyMetadata`
- `factorKeyMetadata.message === "KEY_NOT_FOUND"`
- `factorKeyMetadata.message === "SHARE_DELETED"`

if i recall correctly, i added this case because of some tests were failing at later point, while they shouldn't have passed initialization in the first place (because share was deleted).

Reply:
so in other words, if i recall correctly, there was a case where we had metadata, but share was deleted. or this can't be?

Reply:
mistaken.
thought it was write metadata